lib/TPP/RewriteToBatchReduceGemm.cpp

```diff
@@ -121,6 +116,11 @@ static LogicalResult checkVNNIAccessPatterns(linalg::LinalgOp linalgOp) {
   // Expected access patterns of BRGEMM.
   expectedMaps = infer(
       {{r1, p4, r3}, {r1, r3.floorDiv(*blockingFactor), p5, r2}, {p4, p5}});
+  auto secondOperand = linalgOp.getDpsInputOperands()[1];
+  assert(
```

Comment:
I think this assert should not be here. You are checking the maps in line 124.

Reply:
This is to check that the observed blocking factor is exactly equal to the expected blocking factor.

Reply:
Can you at least common up the shape, so that the assert isn't multi-line?